### PR TITLE
Fix ghc and hlint warnings

### PR DESCRIPTION
**Pull request by jhrcek**

### CODE REVIEW

File: lsp-test/src/Language/LSP/Test/Files.hs [hunk 1]
```diff
@@ -83,18 +83,13 @@ mapUris f event =
         swapDocumentChangeUri (InR (InR (InL renameFile))) = InR $ InR $ InL $ L.newUri .~ f (renameFile ^. L.newUri) $ renameFile
         swapDocumentChangeUri (InR (InR (InR deleteFile))) = InR $ InR $ InR $ swapUri id deleteFile
      in e
-          & L.changes . _Just %~ swapKeys f
+          & L.changes . _Just %~ M.mapKeys f
           & L.documentChanges . _Just . traversed %~ swapDocumentChangeUri
 
-  swapKeys :: (Uri -> Uri) -> M.Map Uri b -> M.Map Uri b
```
Comment (jhrcek) [inline lsp-test/src/Language/LSP/Test/Files.hs:89]: The only non-mechanical change, I just noticed and couldn't resist the simplifcation :smiley:

File: cabal.project [hunk 2]
```diff
@@ -10,12 +10,15 @@ test-show-details: direct
 
 benchmarks: True
 
+package *
+  ghc-options: -Wunused-packages
```
Comment (michaelpj) [inline cabal.project:14]: Hmm. I think we normally put warning options in the cabal files, can we do that here too? Putting them all in `cabal.project` wouldn't be totally unreasonable, it's just inconsistent.

**Review by michaelpj — APPROVED** [review]

File: cabal.project (hunk 2 above)
Comment (jhrcek) [reply cabal.project]: I was just lazy :smile: 
But I agree and I moved it to cabal files + slightly consolidating using `common warnings` stanza and fixed few more warnings.
I hope there's nothing controversial, but will wait few days for your feedback (or let you merge if you think it's ok).

File: lsp-types/lsp-types.cabal [hunk 3]
```diff
@@ -53,7 +57,6 @@ library
     GADTs
     NegativeLiterals
     OverloadedStrings
-    StandaloneKindSignatures
```
Comment (jhrcek) [inline lsp-types/lsp-types.cabal:56]: Removing this, as it's contained in GHC2021

**Comment by michaelpj** [comment]
Great, thanks! Just needs the formatting fixed.

**Comment by jhrcek** [comment]
Fixed. Also noticed that I don't have right to merge (and I don't necessarily want them) so leaving merging up to you.

**Pull request merged**